### ISSUE TITLE
Keyboard interrupt can wipe config

### ISSUE THREAD

**Issue by TheDcoder** [derrod/legendary#289]
## Platform

Operating system and version: Arch Linux rolling
Legendary version (`legendary -V`): 0.20.6 - "A Red Letter Day"

## Expected Behavior

Nothing.

## Current Behavior

`config.ini` could potentially be wiped if a keyboard interrupt (Ctrl+C) is sent while downloading:
```
TheDcoder@arch ~> legendary install 9d2d0eb64d5c44529cece33fe2a46482 --no-install --ignore-free-space
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Preparing download...
[Core] INFO: Parsing game manifest...
[Core] INFO: Install path: /mnt/tank/games/Epic Games/GTAV
[cli] INFO: Install size: 100586.26 MiB
[cli] INFO: Download size: 98862.86 MiB (Compression savings: 1.6%)
[cli] INFO: Reusable size: 0.00 MiB (chunks) / 0.00 MiB (unchanged / skipped)
[cli] INFO: Installation requirements check returned the following results:
[cli] WARNING: Potentially not enough available disk space! 36460.91 MiB < 100586.26 MiB
[cli] WARNING: This game is not marked for offline use (may still work).
[cli] INFO: Downloads are resumable, you can interrupt the download with CTRL-C and resume it using the same command later on.
Do you wish to install "Grand Theft Auto V"? [Y/n]: y
[DLManager] INFO: Download Manager running with process-id: 35761
[DLManager] INFO: Starting download workers...
[DLManager] INFO: Starting file writing worker...
[DLManager] INFO: = Progress: 0.00% (0/101361), Running for 00:00:00, ETA: 00:00:00
[DLManager] INFO:  - Downloaded: 0.00 MiB, Written: 0.00 MiB
[DLManager] INFO:  - Cache usage: 1.0 MiB, active tasks: 0
[DLManager] INFO:  + Download	- 0.00 MiB/s (raw) / 0.00 MiB/s (decompressed)
[DLManager] INFO:  + Disk	- 0.00 MiB/s (write) / 0.00 MiB/s (read)
(...snip...)
[DLManager] INFO: = Progress: 0.25% (257/101361), Running for 00:00:43, ETA: 04:42:35
[DLManager] INFO:  - Downloaded: 138.95 MiB, Written: 166.35 MiB
[DLManager] INFO:  - Cache usage: 36.0 MiB, active tasks: 32
[DLManager] INFO:  + Download	- 2.55 MiB/s (raw) / 3.00 MiB/s (decompressed)
[DLManager] INFO:  + Disk	- 1.00 MiB/s (write) / 0.00 MiB/s (read)
^C[cli] INFO: Command was aborted via KeyboardInterrupt, cleaning up...
Traceback (most recent call last):
  File "legendary/cli.py", line 1375, in <module>
  File "legendary/cli.py", line 1367, in main
  File "legendary/core.py", line 1218, in exit
[DLManager] WARNING: Immediate exit requested!
  File "legendary/lfs/lgndry.py", line 265, in save_config
  File "configparser.py", line 912, in write
KeyboardInterrupt
[35749] Failed to execute script cli
[DLWorker 15] WARNING: Immediate exit requested, quitting...
[DLWorker 10] WARNING: Immediate exit requested, quitting...
[DLWorker 13] WARNING: Immediate exit requested, quitting...
[DLWorker 14] WARNING: Immediate exit requested, quitting...
[DLWorker 16] WARNING: Immediate exit requested, quitting...
[FileWorker] WARNING: Immediate exit requested, quitting...
[DLWorker 5] WARNING: Immediate exit requested, quitting...
[DLWorker 1] WARNING: Immediate exit requested, quitting...
[DLWorker 3] WARNING: Immediate exit requested, quitting...
[DLWorker 7] WARNING: Immediate exit requested, quitting...
[DLWorker 4] WARNING: Immediate exit requested, quitting...
[DLWorker 11] WARNING: Immediate exit requested, quitting...
[DLWorker 8] WARNING: Immediate exit requested, quitting...
[DLWorker 6] WARNING: Immediate exit requested, quitting...
[DLWorker 9] WARNING: Immediate exit requested, quitting...
[DLWorker 12] WARNING: Immediate exit requested, quitting...
[DLWorker 2] WARNING: Immediate exit requested, quitting...
multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 1 leaked shared_memory objects to clean up at shutdown
```

The culprit could be the `save_config` as seen in the stack trace, I had `install_path` configured but it was gone after I ran this.

## Steps to Reproduce

1. Initiate a download
2. Pray to god that you hit Ctrl+C at the correct moment
3. Press Ctrl+C

**Comment by TheDcoder** [comment]
I managed to reproduce the bug again and can confirm that write is being attempted, but this time I had write permissions removed for `config.ini`:
```
Traceback (most recent call last):
  File "legendary/cli.py", line 1375, in <module>
  File "legendary/cli.py", line 1367, in main
  File "legendary/core.py", line 1218, in exit
  File "legendary/lfs/lgndry.py", line 264, in save_config
PermissionError: [Errno 13] Permission denied: '/home/wineuser/.config/legendary/config.ini'
```

**Comment by derrod** [comment]
This is *mostly* fixed in 0e6b63d1a264f78db123e43b952d5eb9e43ef709 which skips writing the config if it hasn't changed. In case something like selective downloading was used the config would still be written however (although I'm planning to move that to the moment it's actually changed rather than on exit).

I'm not sure why the CTRL+C also interrupts the config writing rather than just the downloader process, I also have not seen this issue ever before.

**Comment by TheDcoder** [comment]
> I'm not sure why the CTRL+C also interrupts the config writing

I am not an experienced python developer, but perhaps the interrupt "event" is also propagating to the config parser because it isn't being caught?
